test/test_aggregate_sensors.py

```diff
@@ -0,0 +1,180 @@
+################################################################################
+# Copyright (c) 2024, National Research Foundation (SARAO)
+#
+# Licensed under the BSD 3-Clause License (the "License"); you may not use
+# this file except in compliance with the License. You may obtain a copy
+# of the License at
+#
+#   https://opensource.org/licenses/BSD-3-Clause
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+################################################################################
+
+"""Tests for :mod:`katsdpcontroller.aggregate_sensors`."""
+
+import re
+from typing import List
+from unittest import mock
+
+import pytest
+from aiokatcp import Reading, Sensor, SensorSet
+
+from katsdpcontroller.aggregate_sensors import LatestSensor, SumSensor, SyncSensor
+
+
+@pytest.fixture
+def target() -> SensorSet:
+    sensors = SensorSet()
+    # Add an unrelated sensor to verify name_regex behaviour
+    sensors.add(Sensor(int, "ignore-me", "", initial_status=Sensor.Status.NOMINAL, default=123))
+    return sensors
+
+
+@pytest.fixture
+def int_children() -> List[Sensor[int]]:
+    sensors = [Sensor(int, f"int-child-{i}", "") for i in range(4)]
+    sensors[0].set_value(3, timestamp=1234567890.0)
+    sensors[1].set_value(4, timestamp=1234567891.0, status=Sensor.Status.WARN)
+    sensors[2].set_value(5, timestamp=1234567892.0)
+    sensors[3].set_value(6, timestamp=1234567893.0, status=Sensor.Status.UNREACHABLE)
```

Comment:
As discussed just now, please do add a one-liner here reminding the reader that sensor status defaults to `NOMINAL`.